Magento2/Sniffs/Legacy/CodingStandardsIgnoreAnnotationUsageSniff.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento2\Sniffs\Legacy;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+class CodingStandardsIgnoreAnnotationUsageSniff implements Sniff
+{
+    private const CODING_STANDARDS_IGNORE_FILE = '@codingStandardsIgnoreFile';
+
+    private const WARNING_CODE = self::CODING_STANDARDS_IGNORE_FILE . ' annotation must be avoided. ';
+
+    private const WARNING_MESSAGE =
+        self::WARNING_CODE
+        . 'Use codingStandardsIgnoreStart/codingStandardsIgnoreEnd to suppress code fragment '
+        . 'or use codingStandardsIgnoreLine to suppress line. ';
+
+    /**
+     * @inheritDoc
+     */
+    public function register(): array
+    {
+        return [
+            T_OBJECT_OPERATOR
```

Comment:
Are you sure about this operator? I think you should be looking for `T_ASPERAND` (the @ char) or `T_COMMENT`